### PR TITLE
9856 dvui dev guide

### PR DESCRIPTION
**Pull request by vebarina**

### CODE REVIEW

File: docs/dvui/guides.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    },
+    plugins: [
+        new VueLoaderPlugin()
+    ],
+};
+```
+
+##### HTML
+
+In the html we should load the script we created with the webpack and customise the variables we want. Please note that customising the variables is optional and that, as previously mentioned, any variable in the `.env` will have priority from the variables defined in the `index.html`.
```
Comment (mexicat) [inline docs/dvui/guides.md:54]: Capitalize "HTML". "The webpack" -> "webpack".

File: docs/dvui/guides.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+The following example includes the custom variables (Document id, Category id, User Token, Locale), but it is not necessary to add these.
```
Comment (mexicat) [inline docs/dvui/guides.md:56]: Capitalize "ID". "Necessary" -> "mandatory".

File: docs/dvui/guides.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+If your application is developed using technologies other than Vue (for example React or Angular), you can build the Document Validation UI app into a package, as described in our [configuration example](#configure-the-app), and import it to your existing project, or install it as an npm package.
+
+#### Run in React
+
+You can find a full example on using the Document Validation UI in a React application [here](https://medium.com/@pateldhruv020/using-vue-component-in-react-9161f30d29a0).
+
+#### Run in Angular
+
+In [this link](https://www.ais.com/using-a-vue-component-in-an-angular-app/) you will see a step by step on how to use the Document Validation UI in an Angular application.
```
Comment (mexicat) [inline docs/dvui/guides.md:114]: "In this link" -> "at this link".

File: docs/dvui/guides.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+### Iframe integration
+
+#### HTML iframe as Public Document
+
+You can mark your Documents as public. Marking Documents as public will generate a read-only, publicly accessible view that does not require authentication. This allows you to share a link to the Document and its extracted data, or embed it in another website. See [Read Only mode](./modes.md/#read-only-mode) for reference.
```
Comment (mexicat) [inline docs/dvui/guides.md:120]: Mmm, maybe "publicly accessible URL" instead of "publicly accessible view" is clearer.

File: docs/dvui/index.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+![document_thumbnails.svg](./images/document_thumbnails.svg)
+
+This is a simple component that loads the Document images and displays them in a vertical list to act as a thumbnail selector. It is fully clickable and it updates its state automatically.
+
+### Document Page
+
+![document_page.svg](./images/document_page.svg)
+
+This component is responsible for drawing the Document and handling any interaction with it. The main thing is that it uses [Konva JS](https://konvajs.org/docs/vue/index.html) to render a canvas element that has the Document image as background and shapes on top. Most of the shapes are Annotations from the Konfuzio API.
```
Comment (mexicat) [inline docs/dvui/index.md:28]: "The main thing is that it uses" -> "it uses".

File: docs/dvui/modes.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+.. meta::
+:description: description of the two possible modes to run the app
+
+# Read Only Mode vs Full Mode
+
+The Document Validation UI can be configured to be run as Read Only or Full Mode:
+
+## Read Only Mode
+
+This is the default mode of the app. In this mode, you will have a sample Document with annotations that you can only preview. The default API endpoint will be https://app.konfuzio.com and no user account is needed.
```
Comment (mexicat) [inline docs/dvui/modes.md:10]: "Unless configured, it uses the default API endpoint at https://app.konfuzio.com..."

File: docs/dvui/modes.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+## Full Mode
+
+If you want to run the widget in full mode to be able to interact with the Document by editing Annotations, Document pages and other functionalities, you will need to have a user account created (more information in our “Managing users” section). Then, you should generate a user Token, by accessing the [Konfuzio API version 3 Auth Request](https://app.konfuzio.com/v3/swagger/) and making a request with your username and password. If the account exists, then a Token will be generated that you can copy.
```
Comment (mexicat) [inline docs/dvui/modes.md:14]: * "you should generate a user Token, by accessing" -> remove the comma
* "more information in our “Managing users” section" -> can we link it?
* "If the account exists" -> "If the provided credentials are correct"
* "a Token will be generated that you can copy **and add to the env settings**" (maybe with a "see below"?)

File: docs/dvui/modes.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+You will also need a [Document uploaded](https://app.konfuzio.com/v3/swagger/#/documents/documents_create) and a Document id, and will need to be logged in to [Konfuzio](https://app.konfuzio.com/)) before being able to upload the Document. After successfully uploading it, if you want to show it on the Document Validation UI, you can copy the Document id from the URL, as shown in the image below:
+
+![docid.png](./images/docid.png)
+
+You can create an environment variables file `.env` on the root of the repository based on the [`.env.example`](https://github.com/konfuzio-ai/konfuzio-capture-vue/blob/main/.env.example) for specifying various options, especially the API endpoint, the user Token, the Document id and Category to load (optional), and the default language of the app.
```
Comment (mexicat) [inline docs/dvui/modes.md:20]: "You can create" -> I would make it more "you have to create" or maybe something more indirect like "To complete the setup, create an env file..."

Then I think you can only list (maybe as a bullet list) the env options that are actually required and leave the rest in the documentation link (the env example is fine, but don't we have some markdown documentation for that?)

**Review by mexicat — CHANGES_REQUESTED** [review]
Good job!

File: docs/dvui/modes.md (hunk 8 above)
Comment (vebarina) [reply docs/dvui/modes.md]: @mexicat I made the changes, but not sure what you refer to with the "markdown documentation" for the .env. As far as I know we just have the example file, but might be confused as to what you are talking about.

Comment (mexicat) [reply docs/dvui/modes.md]: I was just talking about general documentation about the setting variables in the guides. If they're only in the `.env` example file it's OK like this.

**Review by mexicat — APPROVED** [review]

**Pull request merged**